lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -232,13 +232,15 @@ public void close() {
                     .setKillEvaluator(EvaluatorRuntimeProtocol.KillEvaluatorProto.newBuilder().build())
                     .build();
             sendEvaluatorControlMessage(evaluatorControlProto);
+            this.stateManager.setClosing();
+          } else {
+            this.stateManager.setKilled();
           }
-        } finally {
+        } catch (Exception e) {
```

Comment:
Also, is it really safe here to substitute the `finally` clause with  the `catch` clause ? Shouldn't the  `this.stateManager.setKilled()` always execute after the `try {} `and not only if an exception is caught ?